Maths/IsCompositeNumber.js

```diff
@@ -0,0 +1,30 @@
+/*
+ * Composite number: https://en.wikipedia.org/wiki/Composite_number
+ * function isCompositeNumber
+ * Check if a given number is a composite number or not?
+ * isCompositeNumber(6) // returns true
+ * isCompositeNumber(577) // returns false
+ * isCompositeNumber(2024) // returns true
+ * A composite number is a positive integer that is not prime. In other words, it has a positive divisor other than one or itself.
+ * First few composite numbers are 4, 6, 8, 9, 10, 12, 14, 15, 16, 18, 20, ………
+ * Every integer greater than one is either a prime number or a composite number.
+ * The number one is a unit – it is neither prime nor composite.
+ */
+
+function isCompositeNumber (number) {
+  let i = 1
+  let count = 0
+  while (number >= i) {
```

Comment:
Could you reuse one of the existing sieves here?

Reply:
> Could you reuse one of the existing sieves here?

Hmm, sorry, I don't get what you mean.

Reply:
There are more efficient ways to factorize a number (such as only searching for factors up to the sqrt). Also, as soon as you have found a single factor, you can divide the number by it to find the second factor (if it exists) & exit early.

Reply:
> There are more efficient ways to factorize a number (such as only searching for factors up to the sqrt). Also, as soon as you have found a single factor, you can divide the number by it to find the second factor (if it exists) & exit early.

Oo. I see. I see. Interesting. Do you have any suggestions on how can I improve the algorithm?

Reply:
You could look at using the existing [prime factorization algorithm](https://github.com/TheAlgorithms/JavaScript/blob/master/Maths/PrimeFactors.js).

Reply:
> You could look at using the existing [prime factorization algorithm](https://github.com/TheAlgorithms/JavaScript/blob/master/Maths/PrimeFactors.js).

Okay, I will try to implement and update the algorithm when I am free. Thanks!

Reply:
Updated the algorithm with a different approach.